### PR TITLE
Expose the DegreesToRadians and RadiansToDegrees APIs

### LINKED ISSUE #86402: [API Proposal]: Convert between degrees and radians
### Background and motivation

Most applications performing 2D and 3D spatial calculations need to convert between radians and degrees. For example, Unity includes the constants `Mathf.Deg2Rad` and `Rad2Deg`; Microsoft XNA/MonoGame includes the functions `MathHelper.ToDegrees` and `ToRadians`; and NetTopologySuite inclues the methods `Radians.ToDegrees` and `Degrees.ToRadians`.

### API Proposal

```diff
  namespace System.Numerics {
      public interface ITrigonometricFunctions<TSelf> {
+         public static abstract TSelf ToDegrees(TSelf radians);
+         public static abstract TSelf ToRadians(TSelf degrees);
      }
  }
```

And, if `Math` and `MathF` can still evolve despite the introduction of generic math, it would be nice to also include sugar for these there.

```diff
namespace System
{
    public static class Math
    {
+        public static double ToDegrees(double radians) => radians.ToDegrees();
+        public static double ToRadians(double degrees) => degrees.ToDegrees();
    }
    public static class MathF
    {
+        public static float ToDegrees(float radians) => radians.ToDegrees();
+        public static float ToRadians(float degrees) => degrees.ToDegrees();
    }
}
```

### API Usage

```csharp
var angleInRadians = Math.PI;

// Convert from radians to degrees. Returns 180.0
var angleInDegrees= double.ToDegrees(angleInRadians);

// Convert from degrees to radians. Returns (an aproximation of) π
angleInRadians= double.ToRadians(angleInDegrees);
```

### Alternative Designs

An alternative design was proposed in #38566:

```diff
  namespace System {
      public static class MathF {
+        public const float RadiansToDegrees = 57.295779513f;
+        public const float DegreesToRadians = 0.0174532925f;
    }
```

However, this was rejected in favor of methods:

> Functions will allow a more precise implementation to be provided or for other optimizations to occur, without restricting things.

There was one counterpoint that should still be considered tough:

> A constant could also be used for vectors
> ```cs
> var input = new Vector3(10f, 20f, 30f) * MathF.DegreesToRadians;
> ```

### Risks

* Adding members to an interface is always risky
* There may be some churn and confusion as developers migrate from various APIs in existing libraries, but this is short-term

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #86402

### CODE REVIEW

**Review by bartonjs — APPROVED** [review]

**Comment by GrabYourPitchforks** [comment]
I think the order of operations leads to a loss of precision here. From Sharplab:

```cs
Console.WriteLine(DegToRads1(double.MaxValue));
Console.WriteLine(DegToRads2(double.MaxValue));

static double DegToRads1(double deg) => (deg * Math.PI) / 180.0;
static double DegToRads2(double deg) => deg * (Math.PI / 180.0);
```

Output:

```txt
∞
3.1375664143845866E+306
```

**Comment by tannergooding** [comment]
That’s expected due to the type precision and input domain (as well as general limitations with how we can validly implement the DIM).

The same thing can happen for other pairs of functions as well (like exp and log, cos and acos, etc)

File: src/libraries/System.Private.CoreLib/src/System/Numerics/ITrigonometricFunctions.cs [hunk 1]
```diff
@@ -62,6 +62,54 @@ public interface ITrigonometricFunctions<TSelf>
         /// <returns>The value of <paramref name="degrees" /> converted to radians.</returns>
         static virtual TSelf DegreesToRadians(TSelf degrees)
         {
+            // We don't want to simplify this to: degrees * (Pi / 180)
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/ITrigonometricFunctions.cs:65]: Added a lengthy comment to this API here and pointed the other implementations to it to help avoid people trying to change the algorithm without considering the edge cases.

**Review by GrabYourPitchforks — APPROVED** [review]
Appreciate the comment - thank you! :)

File: src/libraries/System.Private.CoreLib/src/System/Numerics/ITrigonometricFunctions.cs [hunk 2]
```diff
@@ -57,6 +57,73 @@ public interface ITrigonometricFunctions<TSelf>
...
+        {
+            // We don't want to simplify this to: degrees * (Pi / 180)
+            //
+            // Doing so will change the result and in many cases will
+            // cause a loss of precision. The main exception to this
+            // is when the initial multiplication causes overflow, but
+            // if we decide that should be handled in the future it needs
+            // to be more explicit around how its done.
+            //
+            // Floating-point operations are natural imprecise due to
```
Comment (Joe4evr) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/ITrigonometricFunctions.cs:73]:
```suggestion
            // Floating-point operations are naturally imprecise due to
```

**Review by danmoseley — APPROVED** [review]
Great comment

**Pull request merged**